clang/unittests/AST/ASTImporterTest.cpp

```diff
@@ -9919,6 +9919,103 @@ TEST_P(ImportTemplateParmDeclDefaultValue, ImportExistingVarTemplate) {
   testImport(FromLastD);
 }
 
+TEST_P(ImportTemplateParmDeclDefaultValue,
+       ImportParentTemplateDuringNonTypeTemplateParmDecl) {
+  // This wants to provoke that during import of 'Y' in "typename T = Y"
+  // (before this import returns) the later definition of 'X' is imported fully.
```

Comment:
This comment is not accurate, I will remove it.
What exactly should happen:

1. import first `struct X` declaration
3. start import of second `int P`
4. finish import of second `struct X`, this will set `int P` to "inherited"
5. import (visit) function for importing second `int P` returns